openedx/core/djangoapps/discussions/views.py

```diff
@@ -30,6 +31,28 @@
 }
 
 
+class IsStaff(BasePermission):
+    """
+    Check if user is global or course staff
+
+    We create our own copy of this because other versions of this check
+    allow access to additional user roles.
+    > There are many like it but this one is mine.
+    """
+    def has_permission(self, request, view):
+        """
+        Check if user has global or course staff permission
+        """
+        user = request.user
+        if user.is_staff:
+            return True
+        course_key_string = view.kwargs.get('course_key_string')
+        course_key = _validate_course_key(course_key_string)
+        return CourseStaffRole(
+            course_key,
+        ).has_user(request.user)
```

Comment:
This might be a useful addition to `openedx/core/lib/api/permissions.py` as `IsCourseStaff` / `IsGlobalOrCourseStaff`?

Reply:
Agreed. I spent some time refactoring this and `_validate_course_key`, but it kept rabbit-holing due to imports/tests.
I'll try to get that branch working/passing next week, but will plan to merge this as-is for now.